src/metal.cs

```diff
@@ -1156,6 +1257,156 @@ partial interface MTLDevice {
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Unavailable (PlatformName.MacCatalyst)]
+		[NoMac, NoTV, iOS (13, 0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("sparseTileSizeInBytes")]
+		nuint SparseTileSizeInBytes { get; }
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Unavailable (PlatformName.MacCatalyst)]
+		[NoMac, NoTV, iOS (13,0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("newRasterizationRateMapWithDescriptor:")]
+		[return: NullAllowed]
+		IMTLRasterizationRateMap CreateRasterizationRateMap (MTLRasterizationRateMapDescriptor descriptor);
+
+		[Unavailable (PlatformName.MacCatalyst)]
+		[NoMac, NoTV, iOS (13,0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("convertSparseTileRegions:toPixelRegions:withTileSize:numRegions:")]
+		void ConvertSparseTileRegions (IntPtr tileRegions, IntPtr pixelRegions, MTLSize tileSize, nuint numRegions);
+
+		[Unavailable (PlatformName.MacCatalyst)]
+		[NoMac, NoTV, iOS (13,0)]
+		[Advice ("This API is not available when using UIKit on macOS.")]
+		[Export ("convertSparsePixelRegions:toTileRegions:withTileSize:alignmentMode:numRegions:")]
+		void ConvertSparsePixelRegions (IntPtr pixelRegions, IntPtr tileRegions, MTLSize tileSize, MTLSparseTextureRegionAlignmentMode mode, nuint numRegions);
+
+		[TV (13, 0), Mac (10, 15), iOS (13, 0)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("hasUnifiedMemory")]
+		bool HasUnifiedMemory { get; }
+
+		[TV (13,0), Mac (10,15), iOS (13,0)]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("supportsFamily:")]
+		bool SupportsFamily (MTLGpuFamily gpuFamily);
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV, Mac (10, 15)]
+		[Export ("barycentricCoordsSupported")]
+		bool BarycentricCoordsSupported { [Bind ("areBarycentricCoordsSupported")] get; }
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV, Mac (10, 15)]
+		[Export ("supportsShaderBarycentricCoordinates")]
+		bool SupportsShaderBarycentricCoordinates { get; }
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV, Mac (10, 15)]
+		[Export ("peerIndex")]
+		uint PeerIndex { get; }
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV, Mac (10, 15)]
+		[Export ("peerCount")]
+		uint PeerCount { get; }
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV, Mac (10, 15)]
+		[NullAllowed, Export ("counterSets")]
+		MTLCounterSet[] CounterSets { get; }
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV, Mac (10,15)]
+		[Export ("newCounterSampleBufferWithDescriptor:error:")]
+		[return: NullAllowed]
+		MTLCounterSampleBuffer CreateCounterSampleBuffer (MTLCounterSampleBufferDescriptor descriptor, [NullAllowed] out NSError error);
+
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[NoiOS, NoTV, Mac (10,15)]
+		[Export ("sampleTimestamps:gpuTimestamp:")]
+		void SampleTimestamps (nuint cpuTimestamp, nuint gpuTimestamp);
```

Comment:
```suggestion
		void GetSampleTimestamps (nuint cpuTimestamp, nuint gpuTimestamp);
```

However, this binding is wrong, because the parameters are `NSUInteger *`, which are probably C-style arrays of NSUInteger.